qcodes/instrument_drivers/rigol/DG1062.py

```diff
@@ -0,0 +1,182 @@
+"""
+Module for the Rigol DG1062 driver. We have only implemented:
+1) Setting/getting waveforms
+2) Setting/getting the impedance
+3) Setting/getting the polarity
+"""
+
+import logging
+from functools import partial
+from typing import Union, Dict
+
+from qcodes import VisaInstrument, validators as vals
+from qcodes import InstrumentChannel, ChannelList
+
+log = logging.getLogger(__name__)
+
+
+class DG1062Channel(InstrumentChannel):
+    def __init__(self, parent: 'DG1062', name: str, channel: int) ->None:
+        """
+        Args:
+            parent: The instrument this channel belongs to
+            name (str)
+            channel (int)
+        """
+
+        super().__init__(parent, name)
+
+        self.parent = parent
+        self.channel = channel
+
+        # All waveforms except 'DC' and 'ARB' have these parameters
+        default_wave_params = ["freq", "ampl", "offset", "phase"]
+
+        self.waveform_params = {
+            waveform: default_wave_params for waveform in
+            ["HARM", "NOIS", "RAMP", "SIN", "SQU", "TRI", "USER"]
+        }
+
+        self.waveform_params["DC"] = ["freq", "ampl", "offset"]
+        self.waveform_params["ARB"] = ["sample_rate", "ampl", "offset"]
+
+        for param, unit in [
+            ("freq", "Hz"),
+            ("ampl", "V"),
+            ("offset", "V"),
+            ("phase", "deg"),
+            ("sample_rate", "1/s")
+        ]:
+            self.add_parameter(
+                param,
+                unit=unit,
+                set_cmd=partial(self._set_waveform_param, param),
+                get_cmd=partial(self._get_waveform_param, param)
+            )
+
+        self.add_parameter(
+            "waveform",
+            get_cmd=partial(self._get_waveform_param, "waveform")
+            # We set the waveform by calling the 'apply' method with
+            # appropriate parameters
+        )
+
+        self.add_parameter(
+            "impedance",
+            get_cmd=f":OUTPUT{channel}:IMP?",
+            set_cmd=f":OUTPUT{channel}:IMP {{}}",
+            unit="Ohm",
+            vals=vals.Ints(min_value=1, max_value=10000)
+        )
+
+        self.add_parameter(
+            "polarity",
+            get_cmd=f":OUTPUT{channel}:GAT:POL?",
+            set_cmd=f":OUTPUT{channel}:GAT:POL {{}}",
+            vals=vals.OnOff(),
+            val_mapping={1: 'POSITIVE', 0: 'NEGATIVE'},
+        )
+
+    def apply(self, **kwargs: Dict) ->Union[None, Dict]:
+        """
+        Apply a waveform on the channel
+
+        Example:
+        >>> gd = DG1062("gd", "TCPIP0::169.254.187.99::inst0::INSTR")
+        >>> gd.channels[0].apply(waveform="SIN", freq=1E3, ampl=1.0, offset=0, phase=0)
```

Comment:
Is there a good way to know the valid arguments. Perhaps even just a ref to the relevant page in a manual

Reply:
The valid arguments depend on the waveform, which makes this a bit tricky. If you give wrong arguments I raise a ValueError and list the arguments expected. See the example notebook. 

Perhaps I can build an alternate interface which looks something like: 
```python
gd.channel[0].apply.sin(freq, ampl, offset, phase)
```
The advantage of that is that we can then have proper doc strings which the user can refer to.

Reply:
![image](https://user-images.githubusercontent.com/10454668/39702076-d933eee2-5203-11e8-8e63-30b93a5e49b5.png)

Reply:
sweet!